File: cafeland-club-frontend/src/app/features/semanas/services/semana.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Semana, SemanaGuardarRequest } from '../models/semana.model';

@Injectable({ providedIn: 'root' })
export class SemanaService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = '/api/semanas';

  listarTodas(): Observable<Semana[]> {
    return this.http.get<Semana[]>(this.apiUrl);
  }

  obtenerPorId(id: number): Observable<Semana> {
    return this.http.get<Semana>(`${this.apiUrl}/${id}`);
  }

  registrar(request: SemanaGuardarRequest): Observable<Semana> {
    return this.http.post<Semana>(this.apiUrl, request);
  }

  actualizar(id: number, request: SemanaGuardarRequest): Observable<Semana> {
    return this.http.put<Semana>(`${this.apiUrl}/${id}`, request);
  }
}

File: cafeland-club-backend/src/main/java/com/cafeland/club/service/SemanaService.java
package com.cafeland.club.service;

import com.cafeland.club.dto.semana.SemanaActualizarRequest;
import com.cafeland.club.dto.semana.SemanaCrearRequest;
import com.cafeland.club.dto.semana.SemanaResponse;

import java.util.List;
import java.util.Optional;

public interface SemanaService {

    List<SemanaResponse> listarTodas();

    Optional<SemanaResponse> obtenerPorId(Long id);

    SemanaResponse registrar(SemanaCrearRequest request);

    Optional<SemanaResponse> actualizar(Long id, SemanaActualizarRequest request);
}
